services/venice-router/src/test/java/com/linkedin/venice/router/api/TestVeniceVersionFinder.java

```diff
@@ -318,8 +320,11 @@ public void returnsCurrentVersionWhenItIsTheOnlyOption() {
 
     String firstVersionKafkaTopic = Version.composeKafkaTopic(storeName, firstVersion);
 
-    Assert.assertEquals(versionFinder.getVersion(storeName, request), firstVersion);
+    Assert.assertEquals(versionFinder.getVersion(storeName, request), Store.NON_EXISTING_VERSION);
     Assert.assertNull(compressorFactory.getVersionSpecificCompressor(firstVersionKafkaTopic));
+
+    doReturn(true).when(compressorFactory).versionSpecificCompressorExists(firstVersionKafkaTopic);
+    Assert.assertEquals(versionFinder.getVersion(storeName, request), firstVersion);
```

Comment:
Now that this test behavior has changed, can you also update the comment (and test name if needed) ?
```
    // When the router doesn't know of any other versions, it will return that version even if dictionary is not
    // downloaded.
    // If the dictionary is not downloaded by the time the records needs to be decompressed, then the router will return
    // an error response.
```